### DISCUSSION TITLE
How to see list of modules compiled?

### DISCUSSION THREAD

**Discussion by tinleym** [vercel/next.js#50877] [Help] [answered]
### Summary

I'm getting very slow build times in dev, and in the console I'm seeing a lot modules being compiled: `event compiled client and server successfully in 21.8s (2214 modules)`. 

I'm using modularizeImports for my icon library, which has helped, but I'd like to see a list of all 2k+ compiled modules so I might hopefully be able to also address those.

Does anybody know how I might approach this?

**Comment by SamKomesarook** [comment, marked as answer]
You can use the [next bundle analyzer](https://github.com/vercel/next.js/tree/canary/packages/next-bundle-analyzer) to analyse the sizes of the different modules in your bundle.

**Reply by tinleym** [reply]
Thank you!

**Reply by MonstraG** [reply]
That will show me information about the entire app. How can I see what were the modules for a specific page?

**Reply by zeptonaut** [reply]
Once you're in the main "Next bundle analyzer" UI, click the arrow in the upper left to show the filter UI, then click into the "Filter to initial chunks" and select the specific page (entrypoint) that you're trying to debug.

https://github.com/user-attachments/assets/5835593a-937c-43e9-b9a8-027ebb7fd7e9
